csrc/scheduler/matmul_utils.cpp

```diff
@@ -307,32 +309,63 @@ std::string getMatmulCompileTimeRejectReason(Fusion* fusion) {
   // 1. check if there is exactly one MmaOp defined in the fusion
   // 2. check if MmaOp inputs match any of supported inputs layout
   // 3. check if fusion represents expressions that are recognized by matmul
-  //    scheduler
+  // scheduler In addition to the above, it may be the case that there is no mma
+  // op, but a mul-sum pair which may be replaced by a mma op. In that case we
+  // run a pass through the IR to find such a pair of mul-sum, and the
+  // properties of that pair. We then test if the mma op that would replace the
+  // mul-sum pair can be scheduled.
 
   // #1
   auto mma_exprs = ir_utils::getOpsOfType<MmaOp>(fusion);
+  // Initializing the machinery to check if there's a Mul-Sum pair
```

Comment:
How about changing the API of the `mma_utils::CombineMulSum` in a way to hide implementation details, the fact that there is `mul-sum` pattern or `MmaOp`? In future it will make it much simpler to remote `MmaOp` from handling.

Something like this:
```
// The analysis is based on mul-sum pair pattern, detected in the provided fusion
//  definition. If detected and properties of an instance of such pattern are valid
//  then it will be later replaced with MmaOp in fusion definition.
// For the time being a direct instance of MmaOp is also accepted and handled by
//  the analysis.
 
// The plan:
// 1. check if there is exactly one mul-sum pattern in the fusion, an instance of
//     MmaOp works too
// 2. check if mul-sum inputs match any of supported inputs layout
// 3. check if fusion represents expressions that are recognized by matmul scheduler

// #1
const mma_utils::CombineMulSum combiner(fusion);

// of the record notes: isValid() will check if fusion contains mul-sum pair (1st step) or if there is a MmaOp (1nd step) 
if (!combiner.isValid()) {
  return "Failed to find mul-sum pattern in the provided fusion";
  // of the record notes: there can be `getValidityError()` which will provide more detailed explanation why fusion does not meet requirements, e.g. no pattern / no MmaOp
}
const auto mul_sum_candidates = combiner.generateMulSumCanidates();
if (mul_sum_candidates.size() != 1) {
  std::stringstream ss;
  ss  << "Matmul scheduler supports fusions with only single mul-sum pattern, got: " << mul_sum_candidates.size();
  return ss.str();
}

// #2 
{
  // of the record notes: combiner returns inputs/outputs for any supported source, be it mul-sum pattern or MmaOp
  //    this should be done during creation of mma_utils::CombineMulSum
  const auto input_layout_opt = mma_utils::getMmaLayout(fusion, mul_sum_candidates.front().insouts());
  if (!input_layout_opt.isValid()) {
    return input_layout_opt.getErrorMsg();
  }
}

// #3
{
  auto support_status = isMatmulFusionDefinitionSupported(fusion, mul_sum_candidates.front().insouts());
  if (!support_status.empty()) {
    return support_status;
  }
}
```